pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -3054,6 +3054,11 @@ func (tmpl *Template) HasOutput() bool {
 	return tmpl.Container != nil || tmpl.ContainerSet.HasContainerNamed("main") || tmpl.Script != nil || tmpl.Data != nil || tmpl.HTTP != nil || tmpl.Plugin != nil
 }
 
+// HasArguments returns true if the template can have arguments.
+func (tmpl *Template) HasArguments() bool {
```

Comment:
I'd prefer if this was `CanHaveArguments` as it's slightly misleadingly named as is.